mlflow/xgboost.py

```diff
@@ -286,6 +288,24 @@ def autolog(importance_types=["weight"]):  # pylint: disable=W0102
     import xgboost
     import numpy as np
 
+    def __init__(self, *args, **kwargs):
+        data = args[0] if len(args) > 0 else kwargs["data"]
```

Comment:
what happens if `len(args) == 0` and `data` is not in `kwargs` (e.g., user error)?